spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -655,8 +655,23 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           (builder, mathExpr) => builder.setIntegralDivide(mathExpr))
 
         if (divideExpr.isDefined) {
+          val childExpr = if (dataType.isInstanceOf[DecimalType]) {
+            // check overflow for decimal type
+            val builder = ExprOuterClass.CheckOverflow.newBuilder()
+            builder.setChild(divideExpr.get)
+            builder.setFailOnError(false)
```

Comment:
Is it always `false` even for ANSI mode?

Reply:
> Is it always `false` even for ANSI mode?

Sorry, I mistakenly thought that comet does not support ansi mode